sdk/core/core-lro/test/lro.spec.ts

```diff
@@ -2549,48 +2650,100 @@ matrix(
                 method: "GET",
                 path: pollingPath,
-                body: `{ "status": "running"}`,
+                body: `{ "status": "running" }`,
                 status: 200,
               }),
               {
                 method: "GET",
                 path: pollingPath,
-                body: `{ "status": "succeeded"}`,
+                body: `{ "status": "succeeded" }`,
                 status: 200,
               },
             ],
             throwOnNon2xxResponse,
             implName,
             updateState: () => {
               pollCount++;
+              if (pollCount === 10) {
+                abortController.abort();
+              }
             },
           });
-          const abortController = new AbortController();
           await poller.poll();
-          abortController.abort();
           assert.equal(pollCount, 1);
-          const promise = poller.pollUntilDone();
-          poller.stopPolling();
+          const promise = poller.pollUntilDone({
+            abortSignal: abortController.signal,
+          });
           await assertError(promise);
-          /**
-           * There is a behavior difference in how each poller is being stopped.
-           * TODO: revisit this if it becomes an issue.
-           */
-          assert.equal(pollCount, implName === "createPoller" ? 2 : 1);
-          assert.isFalse(poller.isDone());
+          assert.equal(pollCount, 10);
+          assert.isFalse(poller.isDone);
         });
       });
       describe("general behavior", function () {
+        it("awaitting poller would return result directly", async () => {
+          const bodyObj = {
+            properties: { provisioningState: "Succeeded" },
+            id: "100",
+            name: "foo",
+          };
+          const retResult = {
+            ...bodyObj,
+            statusCode: 200,
+          };
+          const pollingPath = `pollingPath`;
+          let pollCount = 0;
+          const pollingRoutes = [
+            {
+              method: "POST",
+              status: 202,
+              headers: {
+                "Operation-Location": pollingPath,
+              },
+            },
+            ...Array(10).fill({
+              method: "GET",
+              path: pollingPath,
+              body: `{ "status": "running" }`,
+              status: 200,
+            }),
+            {
+              method: "GET",
+              path: pollingPath,
+              body: JSON.stringify(bodyObj),
+              status: 200,
+            },
+          ];
+          const poller = createTestPoller({
+            routes: pollingRoutes,
+            throwOnNon2xxResponse,
+            implName,
+            updateState: () => {
+              pollCount++;
+            },
+          });
+          assert.isUndefined(poller.operationState);
+          assert.isUndefined(poller.result);
+          assert.equal(pollCount, 0);
+          const result = await poller;
```

Comment:
what happens if you do:

```ts
const promise = poller;
promise.catch();
const result = await poller;
```

Reply:
i added new cases here: https://github.com/Azure/azure-sdk-for-js/pull/28213/commits/3f5ba0fbe549fdf719ca5a21d88dfec1166c9aa2